mathics/builtin/specialfns/bessel.py

```diff
@@ -667,7 +667,17 @@ class SphericalBesselJ(_Bessel):
 
     attributes = ("Listable", "NumericFunction", "Protected")
 
-    rules = {"SphericalBesselJ[n_, z_]": "Sqrt[Pi / 2] / Sqrt[z] BesselJ[n + 0.5, z]"}
+    rules = {
+        "SphericalBesselJ[n_, z_]": """
+             Times[Rational[1, 2], 
+                   Power[Pi, Rational[1, 2]], 
+                   Power[2, Rational[1, 2]], 
+                   Power[z, Rational[-1, 2]], 
+                   BesselJ[Plus[Rational[1, 2], n], z]
+             ]
+             """,
+        "N[SphericalBesselJ[n_,z_]]": "Sqrt[Pi/2]/Sqrt[z] * BesselJ[N[1/2+n], N[z]]",
```

Comment:
I detect a pattern here. There is a part of me that feels like a whole bunch of rules like this should be collected together into an autoload file with _comments_ describing why they are useful. 

And then the the root cause is fixed, we would be in a better position to go relook at stuff like this in one place.

Reply:
The funny thing is that this seems to be useful in the current implementation. Let's keep this PR as a draft, since this does not have a very huge impact on performance.